packages/core/content-type-builder/server/src/services/components.ts

```diff
@@ -86,13 +86,60 @@ export const editComponent = async (
   return updatedComponent;
 };
 
-export const deleteComponent = async (uid: Internal.UID.Component) => {
-  const builder = createBuilder();
+export const deleteComponent = async (deleteUid: Internal.UID.Component) => {
+  const failedDeletes: any[] = [];
+
+  const deletedComponent = await strapi.db.transaction(async ({ trx }) => {
+    const builder = createBuilder();
+
+    const models = [...builder.contentTypes.entries(), ...builder.components.entries()];
+
+    // Find models that include an attribute with this component as a target and delete references to it from the db
+    for (const [modelUid] of models) {
+      const metadata = strapi.db.metadata.get(modelUid);
+
+      const matchingAttributes = Object.values(metadata.attributes).filter(
+        (attr: any) => attr.target === deleteUid && attr.joinTable?.name
```

Comment:
Just for good measure, should we also check that it is a relation? (also, don't we have an attribute type that we can use that is exported from the DB? If not, not a big deal)